pages/user-interfaces.html

```diff
@@ -55,82 +43,98 @@
+    <li><a href="https://github.com/ejgallego/coq-lsp">coq-lsp</a></li>
+  </ul>
+  <p>
+  Both provides the ability of continuous and asynchronous document
+  checking as well as position-based document navigation (an
+  interaction model introduced in Wenzel's Isabelle/JEdit which
+  departs from the model of splitting the buffer into a validated zone
+  and an unvalidated zone, as found in other Coq user interfaces; see
+  e.g. this <a href="https://github.com/ejgallego/coq-lsp/blob/main/etc/FAQ.md">FAQ</a>
+  for more information, where in the text, "coq-lsp" should be read as
+  "coq-lsp or VsCoq 2" and "VsCoq" should be read as "VsCoq 1").
+  </p>
+  <p>
+  The features provided by the two servers are available to any
+  editor implementing the protocol, e.g. Emacs, Vim, jsCoq, or even another
+  VS Code client. In turn, any new feature eventually added to the
+  servers becomes available for integration by their clients.
+  </p>
 </div>
 
 <div class="frameworklinks">
 <ul>
-<li><a href="refman/practical-tools/coqide.html">CoqIDE</a></li>
-<li><a class="extlink" href="https://jscoq.github.io/">JsCoq</a></li>
+<li><a class="extlink" href="https://github.com/coq-community/vscoq/tree/vscoq1">VsCoq 2</a></li>
+<li><a class="extlink" href="https://github.com/coq-community/vscoq/tree/vscoq1">coq-lsp</a></li>
 </ul>
 </div>
 </div>
 
 <div class="framework">
-<div class="frameworklabel">Experimental / discontinued interfaces</div>
+<div class="frameworklabel">A selected list of experimental or historical interfaces</div>
 <div class="frameworkcontent">
 
 <p>
-There have been many experiments over the years.  Here are some
+Here are some
 additional user interfaces that have stayed at the experimental level
-or whose maintenance has been discontinued and or is currently
+or whose maintenance has either been discontinued or is currently
 uncertain:
 <ul>
+<li> A <a href="https://github.com/EugeneLoy/coq_jupyter/">Jupyter kernel
+for Coq</a> is available if you want to try this interface, for
+instance for pedagogical or readability purposes.  One of the
+advantages of using a Jupyter notebook is that you can save and
+display a selection of intermediate proof steps, which your reader
+will see without the need to reexecute the document.
+</li>
 <li>
 The <a href="http://coqoon.github.io/">Coqoon</a> Eclipse plugin was
 initially based upon
-the <a href="https://bitbucket.org/coqpide/pidetop/src/PIDEtop/">PIDEtop
-Coq plugin</a> (following Wenzel's asynchronous PIDE framework).
-Support for more recent versions of Coq (8.7 and 8.8) relies
-on <a href="https://github.com/coq/coq/blob/master/dev/doc/xml-protocol.md">Coq's
-XML protocol for IDEs</a>.
+the discontinued <a href="https://bitbucket.org/coqpide/pidetop/src/PIDEtop/">PIDEtop
+Coq plugin</a> (following Wenzel's asynchronous PIDE framework and using the
+<a href="https://github.com/coq/coq/blob/master/dev/doc/xml-protocol.md">Coq's
+XML protocol for IDEs</a>).
```

Comment:
The last part of the parenthesis is unclear. Maybe it could be just removed. Otherwise, it could be adapted to be clearer.

Option 1:
```suggestion
Coq plugin</a> (following Wenzel's asynchronous PIDE framework).
```

Option 2:
```suggestion
Coq plugin</a> (following Wenzel's asynchronous PIDE framework), then migrated
to use the <a href="https://github.com/coq/coq/blob/master/dev/doc/xml-protocol.md">XML
protocol of CoqIDE</a>).
```

Reply:
I'm not expert enough to tell. Enrico would.

Reply:
I am confident about the content of the suggested changed based on my reading of the linked webpages.